### PR TITLE
Added Sleep Sort in C++ with markdown file.

### PR DESCRIPTION
**Pull request by achalesh27022003**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2018

### CODE REVIEW

File: C-Plus-Plus/sort/sleep_sort.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+All this multithreading process happens in background and at the core of the OS.
+
+## Complexities
+1. **Time Complexity:** O(NlogN + max(input))
+2. **Space Complexity:** All things are done by internal priority queue in OS. Therefore, auxillary space can be ignored.
+
+## Limitations
+* This algorithm won’t work for negative numbers as a thread cannot sleep for a negative amount of time.
+    
+* Since this algorithm depends on the input elements, so a huge number in the input array causes this algorithm to slow down drastically (as the thread associated with that number has to sleep for a long time). So even if the input array element contains only 2 elements, like- {2, 100000000}, then also we have to wait for a much longer duration to sort.
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/sleep_sort.md]: Remove this

File: C-Plus-Plus/sort/sleep_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      int arg = std::stoi(argv[i]);
+      std::this_thread::sleep_for(std::chrono::seconds(arg));
+      std::cout << argv[i] << std::endl;
+    });
+  }
+ 
+  for (auto& thread : threads) {
+    thread.join();
+  }
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/sleep_sort.cpp]: Add a new line at the EOF

File: C-Plus-Plus/sort/sleep_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+int main(int argc, char* argv[]) {
+  std::vector<std::thread> threads;
+ 
+  for (int i = 1; i < argc; ++i) {
+    threads.emplace_back([i, &argv]() {
+      int arg = std::stoi(argv[i]);
+      std::this_thread::sleep_for(std::chrono::seconds(arg));
+      std::cout << argv[i] << std::endl;
+    });
+  }
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/sleep_sort.cpp]: Put your core Code Logic in a separate function and use the main() function as your Driver Code

File: C-Plus-Plus/math/README.md [hunk 4]
```diff
@@ -1 +0,0 @@
-a
```
Comment (HarshCasper) [inline C-Plus-Plus/math/README.md:1]: These changes make no sense?

File: C-Plus-Plus/other/README.md [hunk 5]
```diff
@@ -1 +1 @@
-a
+
```
Comment (HarshCasper) [inline C-Plus-Plus/other/README.md]: These changes make no sense?

File: C-Plus-Plus/README.md [hunk 6]
```diff
@@ -230,3 +230,5 @@ _add list here_
 - [Sudoku Solver](other/sudoku.cpp)
 - [Swap Two Nibbles](other/SwapTwoNibbles.cpp)
 - [Unique_Number_III](other/Unique_Number_III.cpp)
+- [Count Set no. of bits](other/countsetbits.cpp)
+- [BrianKenighan's Algorithm to count set no. of bits](other/BrianKenighan'sAlgorithm.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: These changes make no sense?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md (hunk 6 above)
Comment (achalesh27022003) [reply C-Plus-Plus/README.md]: These are added as all algorithms which are added are given like this only. That's why I added. If you tell I will remove it.

File: C-Plus-Plus/math/README.md (hunk 4 above)
Comment (achalesh27022003) [reply C-Plus-Plus/math/README.md:1]: Yes they makes sense accidentally wrote a letter so deleted that.

File: C-Plus-Plus/sort/sleep_sort.cpp (hunk 3 above)
Comment (achalesh27022003) [reply C-Plus-Plus/sort/sleep_sort.cpp]: done.

File: C-Plus-Plus/sort/sleep_sort.cpp (hunk 2 above)
Comment (achalesh27022003) [reply C-Plus-Plus/sort/sleep_sort.cpp]: Now no need I think so..Previous code was showing error of scopes etc. I tried new one with multithreading and vector concepts which is working good now.

**Review by achalesh27022003 — COMMENTED** [review]
Please review PR  under DWoC.

File: C-Plus-Plus/README.md (hunk 6 above)
Comment (atarax665) [reply C-Plus-Plus/README.md]: > These are added as all algorithms which are added are given like this only. That's why I added. If you tell I will remove it.

The changes in the README is not corresponding to the file you have added in this PR. You need to update readme for Sleep Sort only.

File: C-Plus-Plus/math/README.md (hunk 4 above)
Comment (atarax665) [reply C-Plus-Plus/math/README.md:1]: Just revert this change.

File: C-Plus-Plus/other/README.md (hunk 5 above)
Comment (atarax665) [reply C-Plus-Plus/other/README.md]: Revert this change @achalesh27022003

File: C-Plus-Plus/sort/sleep_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	for (auto& thread : Threads)
+	{
+		thread.join();
+	}
+}
+
+int main()
+{
+	int NumList[] = {0,4,2,3,5,6,7,9,8,1};
```
Comment (atarax665) [inline C-Plus-Plus/sort/sleep_sort.cpp]: Make changes for dynamic inputs.

File: C-Plus-Plus/sort/sleep_sort.cpp [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	Multithread(NumList);
+	for (auto& Num : Sorted)
+	{
+		std::cout << Num<<std::endl;
+	}
+    return 0;
+}
```
Comment (atarax665) [inline C-Plus-Plus/sort/sleep_sort.cpp:41]: Add sample input/output with complexity of the algorithm.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**